### DISCUSSION TITLE
Changing fuel on fleet #2

### DISCUSSION THREAD

**Discussion by GunnarQuante** [contrailcirrus/pycontrails#86] [Q&A] [answered]
Sorry for circulating back to this: https://github.com/contrailcirrus/pycontrails/discussions/84#discussion-5556505

But even after updating to pycontrails v0.47.0 and pycontrails-bada v0.5.0 there seems to be an issue. When I run the fleet calculation from the snippet above with a SAFBlend, it prints out the following: 

"Jet A-1 / Sustainable Aviation Fuel Blend before CoCiP, with hydrogen content 15.3
Jet A-1 after CoCiP, with hydrogen content 13.8"

The contrail EF changes when compared with a run with JetA  but not as much as I would intuitively expect from a 100% SAFBlend. 

Here's the code I used: 
```

   # Use BADA3 for performance modelling
    bada_model = BADAFlight(met=None, bada_priority=3)
        
    params = {
        "process_emissions": True,
        "verbose_outputs": False,
        "humidity_scaling": ExponentialBoostLatitudeCorrectionHumidityScaling(met=met),
    }
    cocip = Cocip(met=met, rad=rad, params=params, aircraft_performance=bada_model) 
    fuel = SAFBlend(100) # JetA()

    if type(flights_in) == Flight:
        flight = flights_in
        flight.fuel = fuel
        Cocip_out=cocip.eval(flight)
    else: 
        Fleet1= Fleet.from_seq(flights_in)
        Fleet1.fuel = fuel
        print(Fleet1.fuel.fuel_name, 'before CoCiP, with hydrogen content', Fleet1.fuel.hydrogen_content)
        Cocip_out=cocip.eval(Fleet1)
        print(Cocip_out.fuel.fuel_name, 'after CoCiP, with hydrogen content', Cocip_out.fuel.hydrogen_content)
        Cocip_out = Cocip_out.to_flight_list()  
        
    b = []
    for flight in Cocip_out: 
        b.append(np.nansum(flight.data['ef']))
        # print(flight.fuel.fuel_name)
    a = np.nansum(b)
    print ('overall contrail EF of Fleet:', a)
```

**Comment by GunnarQuante** [comment]
I think I found my glitch - the EF values seem to be okay now, it's only the name which changes from prior to after the Cocip evaluation.

**Reply by zebengberg** [reply, marked as answer]
Hi @GunnarQuante,

Thanks for bringing this up again! I was able to reproduce the bug you've found. I agree with your thinking that the predicted EF values are now correct but the fuel attribute on the `Cocip_out` gets lost somewhere.

I've hopefully fixed this in 3cad3758cf4813ee6635e854e13773bc531f28de. Do you install pycontrails through [pypi](https://pypi.org/project/pycontrails/)? If so, I can cut a new release for you. Otherwise, you can just pull down the changes I've made and rerun your simulations.

**Reply by GunnarQuante** [reply]
Hi @zebengberg, 

I just pulled down the new version of cocip.py, now the fuel_name is consistent before and after the cocip evaluation. Thanks a lot!
